### PR TITLE
Make `Conditional` a `SymbolicOp` 

### PR DESCRIPTION
**Pull request by EmilianoG-byte**
**Context:**

The `Conditional` class currently inherits from `Operator`, such that many attributes and methods are missing. 

**Description of the Change:**

By switching to inherit from `SymbolicOp`, the `Conditional` class  now includes some common default behaviour, such as deferring properties to the base class, copying the base class during a shallow copy, and updating the metadata of the base operator during queueing. 

In addition, other relevant representations have been added and tested, e.g.: `matrix`, `adjoint`, `diagonalizing_gates`, and `eigvals`.

As part of the work needed for this PR, new dispatches for `qml.Conditional` and `qml.MeasurementValue` have been added to the `qml.equal()` function.

**Benefits:**
Seamlessly inheritance of properties and basic methods  implemented by `SymbolicOp`, including differing of properties.

**Possible Drawbacks:**
Have to make sure that is not a breaking change as some attributes/methods of `Conditional` are changed.
Need to rethink approach for decomposition of `Conditional` operators.

[sc-61356]

### CODE REVIEW

**Review by astralcai — APPROVED** [review]
LGTM 🎸

**Review by PietropaoloFrisoni — APPROVED** [review]
That's amazing @EmilianoG-byte , thanks!

I just left a couple of minor suggestions :+1:

File: tests/ops/op_math/test_condition.py [hunk 1]
```diff
@@ -362,3 +364,134 @@ def circuit():
         assert tape[1] == mp
         assert isinstance(tape[2], qml.ops.Conditional)
         assert tape[3] == terminal_measurement
+
+
+class TestProperties:
+    """Test Conditional properties"""
```
[the remark this answers is not in the document]
Comment (EmilianoG-byte) [reply tests/ops/op_math/test_condition.py:370]: Thank you for noticing this! I will change it. I am always missing out these little details :D

[the remark this answers is not in the document]
Comment (EmilianoG-byte) [reply tests/ops/op_math/test_condition.py]: Done!

[the remark this answers is not in the document]
Comment (PietropaoloFrisoni) [reply tests/ops/op_math/test_condition.py]: NP, a lot of people do (especially me)

**Pull request merged**